pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -116,6 +117,8 @@ func (r *DNSPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	if !controllerutil.ContainsFinalizer(dnsPolicy, DNSPolicyFinalizer) {
 		if err := r.AddFinalizer(ctx, dnsPolicy, DNSPolicyFinalizer); client.IgnoreNotFound(err) != nil {
 			return ctrl.Result{Requeue: true}, err
+		} else if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
```

Comment:
Why was this required?

Reply:
oh I added this while trying to figure out what was wrong with the e2e tests. I'll remove it now.

Reply:
You keeping this in? Just interested to know what it solves if we do need it, not sure we do this anywhere else.